### PR TITLE
Don't apply magic sorting to BigQuery native queries :imp:

### LINKED ISSUE #2821: Column Ordering
If this is a bug report, please fill in the blanks:

*  I am using the chrome browser.
*  My computer's OS is Mac OS.
*  I'm running Metabase 0.18.0. on Google Container Engine via a docker container
*  My database is BigQuery.
*  My Metabase version is 0.18.0

We're seeing column ordering change in the returned results from 0.17.1 to 0.18.0 - now it seems that columns are returned in ABC order instead of as specified by the query.

I'm going to attempt a downgrade until we have more of an idea of exactly what's up, but wanted to add that info here in case it helps.

### PR DESCRIPTION
**Pull request by camsaul**
One-line fix plus a new unit test.

Fixes #2821

### CODE REVIEW

File: src/metabase/driver/bigquery.clj [hunk 1]
```diff
@@ -351,7 +351,7 @@
         results (if mbql?
                   (post-process-mbql dataset-id table-name results)
                   (update results :columns (partial map keyword)))]
-    (assoc results :annotate? true)))
+    (assoc results :annotate? mbql?)))
```
Comment (salsakran) [inline src/metabase/driver/bigquery.clj]: What's going on here?

I.e. this might be a good place to include a comment ;)

Comment (camsaul) [inline src/metabase/driver/bigquery.clj]: "Annotate" is the post-processing stage of the query processor that adds extra info about columns and sorts the results. It's only supposed to be done for MBQL queries, not native ones

Comment (camsaul) [inline src/metabase/driver/bigquery.clj]: I would add a comment here but that logic is explained in a few other places in the code so I don't think it is necessary to repeat the explanation here

**Comment by camsaul** [comment]
I noticed that BigQuery was missing type information the same way Mongo was (#2732). Going to merge this so I can verify I have a fix for type information for both.

**Pull request merged**